service/spawn.go

```diff
@@ -299,11 +299,37 @@ func (uis *UIServer) modifySpawnHost(w http.ResponseWriter, r *http.Request) {
 		return
 
 	case HostExpirationExtension:
+		if updateParams.Expiration.IsZero() { // set expiration to never expire
+			count, err := host.CountSpawnhostsWithNoExpirationByUser(u.Id)
+			if err != nil {
+				PushFlash(uis.CookieStore, r, w, NewErrorFlash("Error retrieving user spawn host data"))
+				uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrap(err, "Error retrieving user spawn host data"))
+				return
+			}
+			if count >= host.MaxSpawnhostsWithNoExpirationPerUser {
+				msg := fmt.Sprintf("Can only have %d non-expirable spawn hosts", host.MaxSpawnhostsWithNoExpirationPerUser)
+				PushFlash(uis.CookieStore, r, w, NewErrorFlash(msg))
+				uis.LoggedError(w, r, http.StatusBadRequest, errors.New(msg))
+				return
+			}
+			noExpiration := true
+			if err = cloud.ModifySpawnHost(ctx, h, host.HostModifyOptions{NoExpiration: &noExpiration}, env.Settings()); err != nil {
```

Comment:
I think you need to rebase. Sam modified the parameters to this so this won't compile when you merge